sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2644,7 +2644,10 @@ pi_int32 ExecCGCommand::enqueueImpCommandBuffer() {
   // Any non-allocation dependencies need to be waited on here since subsequent
   // submissions of the command buffer itself will not receive dependencies on
   // them, e.g. initial copies from host to device
-  waitForEvents(MQueue, MPreparedDepsEvents, MEvent->getHandleRef());
+  if (!RawEvents.empty()) {
```

Comment:
Worked out that calling `waitForPreparedHostEvents()` here also fixes issues with Arc seen in DPC++ CI, so marking this PR back as draft until i pin down what the complete fix is